include/seqan3/core/simd/view_to_simd.hpp

```diff
@@ -0,0 +1,719 @@
+     * \param matrix The transposed byte matrix from the efficient load procedure.
+     *
+     * \details
+     *
+     * In the efficient load procedure a quadratic byte matrix is first filled and then transposed using e
+     * efficient simd instructions. Depending on the target simd vector type this byte matrix must be unpacked
+     * and the corresponding simd vectors must be assigned to their respective position within the cached
+     * chunk array.
+     */
+    constexpr void split_into_sub_matrices(std::array<max_simd_type, simd_traits<max_simd_type>::length> matrix) const
+    {
+        auto apply_padding = [this] (simd_t const vec)
+        {
+            return (vec == simd::fill<simd_t>(static_cast<uint8_t>(~0))) ? this_view->cached_padding_vector : vec;
+        };
+
+        // Iterate over the rows of the matrix
+        for (uint8_t row = 0; row < static_cast<int8_t>(matrix.size()); ++row)
+        {
+            // split a row into multiple chunks of size `chunk_size`
+            auto chunked_row = unpack(matrix[row]);
+
+            if constexpr (chunked_row.size() == 1)
+            {
+                this_view->cached_simd_chunks[0][row] = apply_padding(std::move(chunked_row[0]));
+            }
+            else // Parse the tuple elements and store them in the cached simd chunks.
+            {
+                static_assert(chunked_row.size() == chunks_per_load, "Expected chunks_per_load many simd vectors.");
+
+                for (uint8_t chunk = 0; chunk < chunks_per_load; ++chunk)  // store chunks in respective cached entries.
+                {
+                    size_t idx = chunk * chunks_per_load + row / chunk_size;
+                    this_view->cached_simd_chunks[idx][row % chunk_size] = apply_padding(std::move(chunked_row[chunk]));
+                }
+            }
+        }
+    }
+
+    /*!\brief Checks if all sequence iterators reached the end.
+     * \tparam indices A non-type template parameter pack over the sequence indices.
+     * \param idx_seq The index sequence.
+     * \returns `true` if all iterators reached the end, otherwise `false`.
+     */
+    template <size_t ...indices>
+    constexpr bool
+    all_iterators_reached_sentinel(std::index_sequence<indices...> const & SEQAN3_DOXYGEN_ONLY(idx_seq)) const noexcept
+    {
+        return (true && ... && (cached_iter[indices] == cached_sentinel[indices]));
+    }
+
+    /*!\brief Convert a single column into a simd vector.
+     * \tparam indices A non-type template parameter pack over the sequence indices.
+     * \param idx_seq The index sequence.
+     * \returns A simd vector filled with the symbols of the sequences at the current position.
+     *
+     * \details
+     *
+     * Converts a single column over the sequences into a simd type. If the end of one sequence was already
+     * reached it will return either a padding value if initialised or `0`.
```

Comment:
```suggestion
     * reached it will return either a padding value if initialised or `0`.
```
What does that sentence mean? I thought the padding_value is defaulted to alphabet_size or a user given one.